### ISSUE TITLE
Add support for procedures on the REPL

### ISSUE THREAD

**Issue by hackaugusto** [0xMiden/miden-vm#637]
# Feature request

At the moment, trying to specify a procedure on the REPL results in the following error:

```
>> proc.test.0 push.1 end
Error running program: AssemblyError(ParsingError("unexpected body termination: invalid token 'proc.test.0'"))
```

Supporting procedures can be useful if one has routines written down and want to use for some interactive testing.

**Comment by bobbinth** [comment]
Yep - that would be a nice feature. We excluded this from the initial version to simplify things. One question was whether we want to limit procedures to single line (and in that case their usefulness may be limited) or if not, what's the best way to handle multi-line input.

**Comment by hackaugusto** [comment]
IMO we should support multi-line input, so that one can copy and paste the procedure from a file.

**Comment by bobbinth** [comment]
Closing in favor of #1746.
